src/logging.rs

```diff
@@ -0,0 +1,61 @@
+extern crate fern;
+
+use log::LevelFilter;
+
+use crate::config::{Configure, UptionConfig};
+
+pub struct Logger {
+    dispatcher: fern::Dispatch,
+}
+
+impl Logger {
+    pub fn new(log_level_filter: LevelFilter) -> Self {
+        let dispatcher = fern::Dispatch::new()
+            .format(|out, message, record| {
+                out.finish(format_args!(
+                    "{}[{}] {}",
+                    chrono::Utc::now().format("[%+]"),
+                    record.level(),
+                    message
+                ))
+            })
+            .level(log_level_filter);
+        Logger { dispatcher }
+    }
+
+    pub fn enable_stdout(self) -> Self {
+        Logger { dispatcher: self.dispatcher.chain(std::io::stdout()) }
+    }
+
+    pub fn enable_log_to_file(self, outfile: &str) -> Self {
+        match fern::log_file(outfile) {
+            Ok(f) => Logger { dispatcher: self.dispatcher.chain(f) },
+            Err(e) => {
+                println!("Logging to file failed: {}", e);
+                self
+            },
+        }
+    }
+
+    pub fn start(self) {
+        self.dispatcher.apply().expect("Firing up logging failed");
+    }
+}
+
+impl Configure for Logger {
+    fn from_config(config: &UptionConfig) -> Self {
+        let logger_config = &config.logger;
+        let logger = Logger::new(logger_config.level_filter);
+
+        let logger = if logger_config.enable_stdout {
+            logger.enable_stdout()
+        } else {
+            logger
+        };
+
+        match &logger_config.outfile {
+            Some(outfile) => logger.enable_log_to_file(&outfile),
+            None => logger
+        }
```

Comment:
Another way to write this would be:
```suggestion
        if let Some(outfile) = &logger_config.outfile {
            return logger.enable_log_to_file(&outfile);
        }
        logger
```
No need to change this unless you think this way would be cleaner.